base/abstractset.jl

```diff
@@ -375,6 +377,27 @@ function issetequal(l, r)
     return issetequal(Set(l), Set(r))
 end
 
+## set disjoint comparison
+"""
+    isdisjoint(v1, v2) -> Bool
+
+Returns whether the collections `v1` and `v2` are disjoint, i.e. whether
+their intersection is empty.
+"""
+function isdisjoint(l, r)
+    function _isdisjoint(l, r)
+        hasfastin(r) && return !any(in(r), l)
+        hasfastin(l) && return !any(in(l), r)
+        haslength(r) && length(r) < FASTIN_SET_THRESHOLD &&
```

Comment:
Just a note, in `issubset` the conversion to `Set` can be done only if `haslength(r)`, here you made the opposite choice... Was it for a specific reason? (I didn't think much about it, but indeed converting to `Set` when length is unknown seems like a better choice to me; what about updating `issubset` then?).

Reply:
Yes, converting to `Set` in general seemed like a good idea to me, with the exception of a fast path when the thing we're iterating over happens to be short.